frontend/src/pages/register/Register.jsx

```diff
@@ -0,0 +1,141 @@
+import React, {useContext, useState} from "react";
+import * as Yup from "yup";
+import {Formik, Form, Field, ErrorMessage} from "formik";
+import "./Register.css";
+import NavigationCard from "components/navigation-card/NavigationCard";
+import {registerUser} from "api/User";
+import {UserContext} from "context/UserContext";
+import {Link, useNavigate} from "react-router-dom";
+
+function Register() {
+  const {setUser} = useContext(UserContext);
+  const [errorMessage, setErrorMesage] = useState();
+  const navigate = useNavigate();
+
+  const handleRegisterSubmit = async (args) => {
+    try {
+      const loggedUser = await registerUser(args);
+      setUser(loggedUser.user);
+      localStorage.setItem("token", loggedUser.token);
+      localStorage.setItem("user", JSON.stringify(loggedUser.user));
+      navigate("/");
+    } catch (exception) {
+      setErrorMesage(exception.response.data);
+    }
+  };
+
+  const RegisterSchema = Yup.object().shape({
+    name: Yup.string()
+      .min(2, "Too Short!")
+      .max(50, "Too Long!")
+      .required("Required"),
+    surname: Yup.string()
+      .min(2, "Too Short!")
+      .max(50, "Too Long!")
```

Comment:
Improve these messages, for example, `Name should contain at least 2 characters` and so on